ros2launch/ros2launch/api/api.py

```diff
@@ -154,11 +186,75 @@ def launch_a_launch_file(*, launch_file_path, launch_file_arguments, debug=False
             launch_arguments=parsed_launch_arguments,
         ),
     ])
+    if secure is not None:
+        # declare keystore so if it is a TemporaryDirectory, it gets cleaned up when out of scope
+        keystore, launch_description = _setup_security(  # noqa: F841
+            launch_description=launch_description,
+            keystore_path=secure,
+            create_keystore=create_keystore,
+        )
     launch_service.include_launch_description(launch_description)
     ret = launch_service.run()
     return ret
 
 
+def _setup_security(
+    *, launch_description: launch.LaunchDescription, keystore_path: str, create_keystore: bool
+) -> Tuple['_Keystore', launch.LaunchDescription]:
+    keystore = _Keystore(keystore_path=keystore_path, create_keystore=create_keystore)
+
+    launch_description = launch.LaunchDescription(
+        [
+            launch.actions.SetEnvironmentVariable(
+                name='ROS_SECURITY_KEYSTORE', value=str(keystore.path)
+            ),
+            launch.actions.SetEnvironmentVariable(name='ROS_SECURITY_STRATEGY', value='Enforce'),
+            launch.actions.SetEnvironmentVariable(name='ROS_SECURITY_ENABLE', value='true'),
+        ]
+        + launch_description.entities
+    )
+
+    return keystore, launch_description
+
+
+class _Keystore:
+    """
+    Object that contains a keystore.
+
+    If a transient keystore is created, contains the temporary directory, assuring
+    it is destroyed alongside the _Keystore object.
+    """
+
+    def __init__(self, *, keystore_path: Union[str, bytes, os.PathLike], create_keystore: bool):
+        if not create_keystore:
+            if keystore_path == '':
+                raise NoKeystoreProvidedError()
+            if not pathlib.Path(keystore_path).exists():
+                raise NonexistantKeystoreError(keystore_path)
+            if not sros2.keystore.is_valid_keystore(keystore_path):
+                raise InvalidKeystoreError(keystore_path)
+
+        # If keystore path is blank, create a transient keystore
+        if keystore_path == '':
+            self._temp_keystore = TemporaryDirectory()
```

Comment:
This also feels ripe for garbage collection. `self._temp_keystore` is never used again, will the collector not pick up on that?

Reply:
self._temp_keystore  is an instance variable, garbage collection won't touch it until the object itself is being collected.

Reply:
Alright.